### PR TITLE
ROB-1395: Fix Coralogix UI permalinks for US2 and other regions

### PR DESCRIPTION
**Pull request by moshemorad**
## What

Fixes broken Coralogix UI permalinks ([ROB-1395](https://linear.app/robusta/issue/ROB-1395/support-coralogix-us2-ui-permalinks)). Permalinks were built as `https://{team_slug}.{domain}`, reusing the configured **API** domain — but the Coralogix team UI lives on a **different hostname in most regions** (per the [official Coralogix domain table](https://coralogix.com/docs/user-guides/account-management/account-settings/coralogix-domain/)).

## The bug (verified against live Coralogix endpoints)

| Configured `domain` | Old permalink host | Result | Correct UI host |
|---|---|---|---|
| `us2.coralogix.com` (**reported case**) | `<team>.us2.coralogix.com` | ❌ TLS cert mismatch | `<team>.app.cx498.coralogix.com` |
| `us1.coralogix.com` | `<team>.us1.coralogix.com` | ❌ TLS cert mismatch | `<team>.app.coralogix.us` |
| `coralogix.us` (from Holmes' own config examples) | `<team>.coralogix.us` | ❌ NXDOMAIN | `<team>.app.coralogix.us` |
| `coralogix.in` (from Holmes' own config examples) | `<team>.coralogix.in` | ❌ NXDOMAIN | `<team>.app.coralogix.in` |
| `eu2.coralogix.com` | `<team>.eu2.coralogix.com` | ⚠️ works via wildcard, but docs-official host is `<team>.app.eu2.coralogix.com` | `<team>.app.eu2.coralogix.com` |

## How it's fixed

- `CORALOGIX_TEAM_HOSTNAME_SUFFIXES` in `holmes/plugins/toolsets/coralogix/utils.py` maps every documented Coralogix domain (current regional forms like `us2.coralogix.com` **and** legacy forms like `cx498.coralogix.com`, `coralogix.us`) to the official team UI hostname suffix. `get_ui_base_url()` normalizes the configured domain (case/whitespace/trailing dot/accidental scheme) and derives the permalink base; unrecognized domains keep the previous `{team_slug}.{domain}` behavior.
- New optional `ui_url` config field overrides the derived UI base URL for non-standard deployments (the second option the ticket asked for).
- Old configs using the deprecated `team_hostname` field also get fixed links.

All official team hostnames were verified live: HTTP 200 with valid TLS, Coralogix app loads and preserves the permalink route/query. Example fixed US2 permalink opened in a real browser loads the Coralogix app over valid TLS, while the old form fails with `SSL: no alternative certificate subject name matches target host name`.

## Tests

- 25 new unit tests: full region/domain matrix (13 domains), domain normalization edge cases, `ui_url` override precedence + normalization, missing `team_slug` fallback, deprecated `team_hostname` compat, and tool-level (`ExecuteDataPrimeQuery._invoke`) permalink assertions for US2.
- Updated `TestCoralogixURLs` in `test_verify_tool_urls.py`, which previously asserted the buggy hostname.
- Full non-LLM suite: **3677 passed, 153 skipped, 0 failed**. Coralogix package coverage: `toolset_coralogix.py` 93%, `utils.py` 81% (`get_ui_base_url` fully covered).

## Docs

- `docs/data-sources/builtin-toolsets/coralogix-logs.md` documents the automatic UI-hostname derivation and the `ui_url` override.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_01UFzbqQfCANSkeNRsvxvTQi

---
_Generated by [Claude Code](https://claude.ai/code/session_01UFzbqQfCANSkeNRsvxvTQi)_

## Summary by CodeRabbit

* **New Features**
  * Coralogix query results now generate clickable links using the correct team UI hostname.
  * Added support for custom Coralogix UI addresses through the optional `ui_url` setting.
  * UI URLs are automatically derived for supported regions, including GOV1, with normalized schemes and trailing slashes.
  * Improved handling for regional and custom Coralogix domains.

* **Documentation**
  * Updated Coralogix configuration guidance with UI URL behavior and customization examples.

* **Tests**
  * Added coverage for regional mappings, custom UI URLs, normalization, and permalink generation.

### CODE REVIEW

File: holmes/plugins/toolsets/coralogix/utils.py [hunk 1]
```diff
@@ -93,6 +134,33 @@ def handle_deprecated_fields(self):
...
+    regions, so the configured API domain cannot be reused verbatim (ROB-1395).
+
+    Resolution order:
+    1. `ui_url`, when configured (explicit override).
+    2. `team_slug` + the official team hostname suffix for the configured domain.
+    3. `team_slug` + the domain as-is, for unrecognized domains.
+    """
+    if config.ui_url:
+        ui_url = config.ui_url.strip().rstrip("/")
+        if not ui_url.startswith(("https://", "http://")):
```
[the remark this answers is not in the document]
Comment (moshemorad) [reply holmes/plugins/toolsets/coralogix/utils.py]: Fixed in 68a5589 — scheme detection is now case-insensitive (`ui_url.lower().startswith(...)`), with a regression test (`test_ui_url_with_uppercase_scheme_is_not_double_prefixed`).

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by moshemorad** [comment]
CI status note — two failing signals on this PR are not caused by this change:

- **Benchmark PR / Benchmark Master**: both fail with an OpenRouter `402` — *"This request requires more credits… limit_source: openrouter_credits"* — the org key's weekly credit limit is exhausted ([job log](https://github.com/HolmesGPT/holmesgpt/actions/runs/34946899283/job/104308341341)). `Benchmark Master` runs master's code and fails identically, so this is infrastructure, not this diff (which only touches Coralogix permalink generation). No code fix exists; it needs credits added to the OpenRouter key. I'm not re-running the jobs since a 402 reproduces by construction and would only burn more credits.
- **License Compliance (FOSSA)**: reports the same "12 issues found" on the master base commit ([`26a3590`](https://github.com/HolmesGPT/holmesgpt/commit/26a35904e846d040ad1e99cc9b21211cb4ab1fbd)) — pre-existing; this PR adds no dependencies.

The unit-test jobs (`test 3.10/3.11/3.12`) are the real gate for this change and are running now; the full suite passes locally (3,677 passed / 0 failed).

---
_Generated by [Claude Code](https://claude.ai/code)_

File: holmes/plugins/toolsets/coralogix/utils.py [hunk 2]
```diff
@@ -84,15 +125,55 @@ def handle_deprecated_fields(self):
...
+
+    if not config.team_slug:
+        return None
+
+    domain = config.domain.strip().lower()
+    domain = domain.removeprefix("https://").removeprefix("http://")
+    domain = domain.strip("/").rstrip(".")
+    team_hostname_suffix = CORALOGIX_TEAM_HOSTNAME_SUFFIXES.get(domain)
+    if team_hostname_suffix is None:
+        if "coralogix" in domain and not domain.startswith("app."):
```
[the remark this answers is not in the document]
Comment (moshemorad) [reply holmes/plugins/toolsets/coralogix/utils.py]: Fixed in b4b0bb0 — the inference now requires a Coralogix-owned apex suffix (`.coralogix.com` or `.coralogixgov.us`, the latter covering future GovCloud regions like the documented `gov1.coralogixgov.us`), with the regression test `test_custom_domain_containing_coralogix_is_not_app_prefixed` for `logs.coralogix-proxy.internal`.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by moshemorad** [comment]
@coderabbitai review

---
_Generated by [Claude Code](https://claude.ai/code)_

File: holmes/plugins/toolsets/coralogix/utils.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (moshemorad) [reply holmes/plugins/toolsets/coralogix/utils.py]: Noting for this thread: superseded by the suffix-based check in b4b0bb0, confirmed above.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: tests/plugins/toolsets/coralogix/test_domain_map_sync.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+ROW_RE = re.compile(
+    r"^\|\s*([a-z0-9.-]+)\s*\|\s*\S+\s*\|[^|]+\|\s*`<team>\.([a-z0-9.-]+)`\s*\|",
+    re.MULTILINE,
+)
+
+
+def test_team_hostname_map_matches_coralogix_docs():
+    """Every region in the official docs table must be mapped, with the same hostname."""
+    try:
+        response = requests.get(DOCS_URL, timeout=30)
```
[the remark this answers is not in the document]
Comment (moshemorad) [reply tests/plugins/toolsets/coralogix/test_domain_map_sync.py:71]: Addressed in 2ee6474, with one deliberate divergence from the proposal:

- **Transient upstream conditions no longer fail the suite** — 429 and 5xx now `pytest.skip` (alongside the existing connection-error/timeout skips), so a docs-site hiccup can't break unrelated PRs.
- **Parsing/comparison logic is now hermetically tested** — `test_docs_table_parsing_and_drift_detection` runs the parser and drift diff on a sample table with no network.
- **The live drift check stays in the default suite** rather than behind an opt-in marker: the PR author explicitly wants CI to fail when the mapping breaks (drift, page moved/404, table format change), and an opt-in check nobody runs wouldn't catch that. With transient statuses skipping, the remaining failure modes are exactly the actionable ones.

---
_Generated by [Claude Code](https://claude.ai/code)_

[the remark this answers is not in the document]
Comment (moshemorad) [reply tests/plugins/toolsets/coralogix/test_domain_map_sync.py:71]: Confirming the author's decision: keeping the live drift check in the default suite is the explicit, on-record requirement for this PR — CI should fail when the mapping actually breaks (drift, moved page, changed table format). With 429/5xx/network conditions skipping, the external dependency can only fail the suite for reasons that require action in this repo. Resolving on that basis.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Review by arikalon1 — APPROVED** [review]
